Added billiardClass, added PWD functions to VerginiSaraceno

clozej opened the pull request.

benokit commented:
Hi Crt,
I will write in English so that anyone can read and understand.

I like what you did but I have some suggestions for improvements:

a) Put PWD logic in a separate file with a meaningful name. The goal is to have clearly separated solvers: vergini-saraceno, pwd,...

b) Separate billiard class definition and drawing utilities into different files.

c) Billiard class should represent geometrical properties of the billiard with functions like:
     - billiard.boundary(numberOfPoints): (int) -> ([double], [double])
     - billiard.boundaryNormal(boundary): ([double], [double]) -> ([double],[double])
     - billiard.area -> double
     - billiard.length -> double
  Solvers should be in separate files. You will use them like:
    - verginiSaraceno.eigenvalues(billiard, k): (Billiard, double) -> [double]
    - ...
 And then we will have procedures independent of the concrete solver:
    - utility.computeSpectrum(billiard, solver, kmin, kmax): (Billiard, Solver, double, double) -> [double]
    - ...

clozej commented:
Hi Benjamin,

I have taken your suggestions into account. 
The plane wave decomposition solvers are now contained in a separate module named planeWaveDecomposition.
 I have also added the previously missing versions of the gradient functions to the verginiSaraceno (VS) module. The functions that evaluate the wave functions and gradient are currently contained in the VS module. I would suggest that we separate them into their own module with a suitable name. This would require some reworking of most of the other modules to correct the import statements.

The billiard class is now reworked from the previous version. I have added a new class called curve. 
The curve class is used to define the border of the billiard. A list of curves is then passed to the billiard object when it is constructed. I have included two examples in the examples folder.

The plotting functions are now separated into a separate module named plottingFunctions.

benokit commented:
Very good Črt. 

A good practice is to have one class definition per file. It would be good to separate curve and billiard classes accordingly. 

Please remove binary and ipynb files from the merge request and put their extensions in the gitignore.

clozej commented:
Separated curve class definition into curveClass.py.
